app/models/mingle/instagram.rb

```diff
@@ -7,45 +7,8 @@ def self.table_name_prefix
   def self.fetch hashtags = Mingle::Hashtag.all
     hashtags = Array(hashtags)
 
-    hashtags.each.map do |hashtag|
-      Instagram.tag_recent_media(hashtag.tag_name_without_hash).collect do |media|
-
-        case media.type
-        when 'image'
-          photo = Photo.find_or_initialize_by photo_id: media.id
-          photo.attributes = {
-            created_at: Time.at(media.created_time.to_i),
-            link: media.link,
-            message: media.try(:caption).try(:text),
-            url: media.images.standard_resolution.url,
-            user_handle: media.user.username,
-            user_id: media.user.id,
-            user_image_url: media.user.profile_picture,
-            user_name: media.user.full_name
-          }
-
-          photo.hashtags << hashtag unless photo.hashtags.include? hashtag
-
-          next if ignore? photo
-
-          photo.save!
-          photo
-        end
-      end.compact
-    end.flatten
+    hashtags.flat_map do |hashtag|
+      PhotosFetcher.new(hashtag).fetch
+    end
```

Comment:
I love flat_map, since i often found myself using map and then flatten. I picked it up recently, but i don't remember where. Thought it was from Rubytapas(Avdi Grimm), but i can't find the video :)